binlog_streamer.go

```diff
@@ -29,6 +29,9 @@ type BinlogStreamer struct {
 	binlogSyncer   *replication.BinlogSyncer
 	binlogStreamer *replication.BinlogStreamer
 
+	DatabaseRewrites map[string]string
+	TableRewrites    map[string]string
```

Comment:
Maybe worth adding a comment linking to this PR as the description is very good. 

Also maybe confusing because this rewrite map **intended to be used** for cases going from target -> source, not source -> target in every other place where we do this. That said, hypothetically we _could_ rewrite the source binlog streamer so this is _theoretically_ generic, but not sure if we should/would.

Reply:
> Also maybe confusing because this rewrite map intended to be used for cases going from target -> source, not source -> target in every other place where we do this. That said, hypothetically we could rewrite the source binlog streamer so this is theoretically generic, but not sure if we should/would.

Yes, agreed this is potentially confusing. I think reworking this is something to be considered as part of a larger rewrite as mentioned in the PR description.